metaworld/envs/mujoco/sawyer_xyz/v2/sawyer_faucet_open_v2.py

```diff
@@ -43,36 +45,32 @@ def model_name(self):
 
     @_assert_task_is_set
     def evaluate_state(self, obs, action):
-        reward, reachDist, pullDist = self.compute_reward(action, obs)
+        (reward, tcp_to_obj, _, target_to_obj, object_grasped,
+         in_place) = self.compute_reward(action, obs)
+
         info = {
-            'reachDist': reachDist,
-            'goalDist': pullDist,
-            'epRew': reward,
-            'pickRew': None,
-            'success': float(pullDist <= 0.05),
+            'success': float(target_to_obj <= 0.11),
+            'near_object': float(tcp_to_obj <= 0.01),
+            'grasp_success': 1.,
+            'grasp_reward': object_grasped,
+            'in_place_reward': in_place,
+            'obj_to_target': target_to_obj,
+            'unscaled_reward': reward,
         }
 
         return reward, info
 
     @property
     def _target_site_config(self):
-        return [
-            ('goal_open', self._target_pos),
-            ('goal_close', np.array([10., 10., 10.]))
-        ]
+        return [('goal_open', self._target_pos),
+                ('goal_close', np.array([10., 10., 10.]))]
 
     def _get_pos_objects(self):
-        knob_center = self.get_body_com('faucetBase') + np.array([.0, .0, .125])
-        knob_angle_rad = self.data.get_joint_qpos('knob_Joint_1')
-
-        offset = np.array([
-            np.sin(knob_angle_rad),
-            -np.cos(knob_angle_rad),
-            0
-        ])
-        offset *= self.HANDLE_LENGTH
+        return self._get_site_pos('handleStartOpen') + np.array(
```

Comment:
Again, this is just a static site pos. Not the real-time position of the object. What was wrong with the code you replaced?

Reply:
The site position used to stay where the initial position of the end of the faucet was.

Reply:
Ok that makes sense, but the obj_pos wasn't returning a site before. It was returning the position of the faucet's tip based on the angle of the joint

Reply:
You know honestly I made the change, things started training with PPO and SAC, and before the change they weren't working. I don't entirely remember why because I worked on these so long ago :(. But the important thing is that it works now without increasing complexity!